src/RobotInterface/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -0,0 +1,554 @@
+                return this->torqueInterface->setRefTorques(nJoints, joints, refs);
+            };
+
+        default:
+            return nullptr;
+        }
+
+        return nullptr;
+    }
+
+    bool setReferences(Eigen::Ref<const Eigen::VectorXd> jointValues)
+    {
+        constexpr std::string_view errorPrefix = "[YarpImplementation::Impl::setReferences] ";
+
+        if(!this->getJointPos())
+        {
+            std::cerr << errorPrefix << "Unable to get the joint position." << std::endl;
+            return  false;
+        }
+
+        const auto worstError = this->getWorstPositionDirectError(jointValues,
+                                                                  this->positionFeedback);
+
+        if (worstError.second > this->positionDirectMaxAdmissibleError)
+        {
+            std::cerr << errorPrefix << "The worst error between the current and the "
+                      << "desired position of the joint named '" << this->axesName[worstError.first]
+                      << "' is greater than " << this->positionDirectMaxAdmissibleError
+                      << " rad. Error = " << worstError.second << " rad." << std::endl;
+            return false;
+        }
+
+        for (const auto& [mode, indeces] : this->desiredJointValuesAndMode.index)
+        {
+            // if indeces vector is empty no joint is controlled with this specific control mode
+            if (indeces.empty())
+                continue;
+
+            if (mode == IRobotControl::ControlMode::Unknown)
+            {
+                std::string error = " The following joints does not have a specified control "
+                                    "mode: ";
+
+                for (const auto& index : indeces)
+                    error += "'" + this->axesName[index] + "' ";
+
+                std::cerr << errorPrefix << error << ". Please set a feasible control mode."
+                          << std::endl;
+
+                return false;
+
+            } else if (mode == IRobotControl::ControlMode::Position)
+            {
+                std::vector<double> refSpeeds(indeces.size());
+                for (int i = 0; i < indeces.size(); i++)
+                {
+                    const auto jointError = std::abs(jointValues[indeces[i]]
+                                                     - this->positionFeedback[indeces[i]]);
+
+                    constexpr double scaling = 180 / M_PI;
+                    refSpeeds[i] = std::max(3.0, scaling * (jointError / this->positioningDuration));
+
+                    this->positionInterface->setRefSpeeds(indeces.size(),
+                                                          indeces.data(),
+                                                          refSpeeds.data());
+                }
+
+                this->startPositionControlInstant = yarp::os::Time::now();
+            }
+
+            // Yarp wants the quantities in degrees
+            double scaling = 180 / M_PI;
```

Comment:
Nitpick: YARP support also prismatic joints, that instead do not want any conversion. We can ignore those, but I suggest at least to leave a comment on this.

Reply:
Hi @GiulioRomualdi , if the conversation is marked as solved but I am not sure why.

Reply:
Ah I see, the comment in https://github.com/dic-iit/bipedal-locomotion-framework/pull/97/files#diff-4a8fdd0d7ae7ceafaa3d549e651d9d0fR29 .